master/internal/api_experiment.go

```diff
@@ -3040,3 +3040,111 @@ func (a *apiServer) DeleteTensorboardFiles(
 
 	return &apiv1.DeleteTensorboardFilesResponse{}, nil
 }
+
+func getPachydermConfig(config map[string]interface{}, experimentID int32) (map[string]interface{}, error) {
+	integrationConfig, ok := config["integration"].(map[string]interface{})
+	// this should not return if proper defaults are used.
+	if !ok {
+		return nil, api.NotFoundErrs("'integration' config for experiment", fmt.Sprint(experimentID), true)
+	}
+
+	pachydermConfig, ok := integrationConfig["pachyderm"].(map[string]interface{})
+	if !ok {
+		return nil, api.NotFoundErrs("'pachyderm' integration config for experiment", fmt.Sprint(experimentID), true)
+	}
+	return pachydermConfig, nil
+}
+
+func (a *apiServer) GetPachydermRepoURL(
+	ctx context.Context, req *apiv1.GetPachydermRepoURLRequest,
+) (resp *apiv1.GetPachydermRepoURLResponse, err error) {
+	// ensure current user has access to this experiment
+	getResp, err := a.GetExperiment(
+		ctx,
+		&apiv1.GetExperimentRequest{
+			ExperimentId: req.ExperimentId,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// extract configs from experiment response
+	pachydermConfig, err := getPachydermConfig(getResp.Config.AsMap(), req.ExperimentId)
```

Comment:
drive by comment: why is this not typed?

Reply:
Good question, this is a remnant of using the data blob and should be adjusted; I'll make sure this is corrected throughout the PR